tests/api/test_api.py

```diff
@@ -9,3 +9,75 @@ def test_get_output_values_prop(self):
+
+
+    # test whether xor model created as autodiff composition learns properly
+    @pytest.mark.parametrize("mode", ['Python',
+                                      pytest.param('LLVMRun', marks=pytest.mark.llvm),
+                                     ])
+    @pytest.mark.parametrize("minibatch_size", [
+        1,
+        2,
+        3,
+        4
+    ])
+    def test_learning_output_shape(self, mode, minibatch_size):
+        '''
+        Tests for correct output from composition.learn
+        Expected: All results from last epoch
+        '''
+        num_epochs=2
+
+        xor_inputs = np.array(  # the inputs we will provide to the model
+            [[0, 0],
+             [0, 1],
+             [1, 0],
+             [1, 1]])
+    
+        xor_targets = np.array(  # the outputs we wish to see from the model
+            [[0],
+             [1],
+             [1],
+             [0]])
+    
+        in_to_hidden_matrix = np.random.rand(2,10)
+        hidden_to_out_matrix = np.random.rand(10,1)
+
+        input_comp = pnl.TransferMechanism(name='input_comp',
+                                       default_variable=np.zeros(2))
+    
+        hidden_comp = pnl.TransferMechanism(name='hidden_comp',
+                                    default_variable=np.zeros(10),
+                                    function=pnl.Logistic())
+
+        output_comp = pnl.TransferMechanism(name='output_comp',
+                                    default_variable=np.zeros(1),
+                                    function=pnl.Logistic())
+
+        in_to_hidden_comp = pnl.MappingProjection(name='in_to_hidden_comp',
+                                    matrix=in_to_hidden_matrix.copy(),
+                                    sender=input_comp,
+                                    receiver=hidden_comp)
+
+        hidden_to_out_comp = pnl.MappingProjection(name='hidden_to_out_comp',
+                                    matrix=hidden_to_out_matrix.copy(),
+                                    sender=hidden_comp,
+                                    receiver=output_comp)
+
+        xor_comp = pnl.Composition()
+
+        backprop_pathway = xor_comp.add_backpropagation_learning_pathway([input_comp,
+                                                                            in_to_hidden_comp,
+                                                                            hidden_comp,
+                                                                            hidden_to_out_comp,
+                                                                            output_comp],
+                                                                            learning_rate=10)
+        target_mech = backprop_pathway.target
+
+        results = xor_comp.learn(inputs={input_comp:xor_inputs,
+                                          target_mech:xor_targets},
+                                  epochs=10,
+                                  minibatch_size=minibatch_size
```

Comment:
this looks like it's missing `bin_execute=mode`, or is there other way to select compiled execution?

Reply:
Ah, I changed the test case for this and in the process I moved it from using AutodiffCompositions -> Standard PNL Compositions.

Since standard PNL doesn't have compiled learning yet, I will have to convert it back to Autodiff - however, the test seemed to fail on the Windows run of GitHub Actions, as `AutodiffComposition` wasn't being successfully imported.

Is there a known fix for this issue?

Reply:
do you mean the 32 bit windows run?
if so, the test should be tagged `@pytest.mark.pytorch` so ti can be skipped if pytorch is not available.